app/services/serialize_offer.rb

```diff
@@ -0,0 +1,64 @@
+class SerializeOffer
+  attr_reader :application_choice, :standard_conditions, :further_condition_attrs
+
+  def initialize(application_choice:, standard_conditions:, further_condition_attrs:)
+    @application_choice = application_choice
+    @standard_conditions = standard_conditions & MakeOffer::STANDARD_CONDITIONS
+    @further_condition_attrs = further_condition_attrs
+  end
+
+  def save
+    @offer = Offer.find_or_create_by(application_choice: application_choice)
+
+    serialize_standard_conditions
+    serialize_further_conditions
+  end
+
+  def conditions
+    further_condition_attrs.values.map { |hash| hash['text'] } + standard_conditions
+  end
+
+private
+
+  def serialize_standard_conditions
+    existing_standard_conditions = @offer.conditions.where(text: MakeOffer::STANDARD_CONDITIONS)
+
+    standard_conditions.each do |text|
+      existing_standard_conditions.find_or_create_by(text: text)
+    end
+    conditions_to_destroy = existing_standard_conditions.where.not(text: standard_conditions)
+    conditions_to_destroy.destroy_all
+  end
+
+  def serialize_further_conditions
+    new_conditions = further_condition_attrs.each_value.map do |params|
+      create_or_update_condition(params)
+    end
+
+    remove_deleted_conditions(new_conditions)
+  end
+
+  def remove_deleted_conditions(new_conditions)
+    existing_conditions_ids = further_condition_attrs.values.inject(new_conditions.map(&:id)) do |conditions, hash|
+      conditions << hash['condition_id']
+    end
+
+    offer_further_conditions.where.not(id: existing_conditions_ids).destroy_all
+  end
+
+  def create_or_update_condition(params)
+    existing_condition = offer_further_conditions.find_by(id: params['condition_id'])
```

Comment:
I think if a condition_id that is not part of the offer gets here, we should be raising an error and not be processing the request at all.

Reply:
Yeah that seems sensible. Note that `condition_id` can be `nil` (when we're making a new condition), and `find` will throw in that case as well as in the case where we get a suspicious id